src/test/java/seedu/address/logic/parser/delivery/DeliveryListCommandParserTest.java

```diff
@@ -148,4 +181,42 @@ public void parse_validStatusAndSort_returnsDeliveryListCommand() {
         CommandParserTestUtil.assertParseSuccess(parser, VALID_DELIVERY_LIST_SHIPPED + VALID_DELIVERY_LIST_SORT_DESC,
             new DeliveryListCommand(DeliveryStatus.SHIPPED, null, null, Sort.DESC));
     }
+
+    @Test
+    public void equals() {
+        DeliveryListCommand deliveryListCommand = new DeliveryListCommand(DeliveryStatus.SHIPPED, null,
+            new Date("2023-12-12"), Sort.ASC);
+
+        // same object -> returns true
+        assertEquals(deliveryListCommand, deliveryListCommand);
+
+        // same values -> returns true
+        DeliveryListCommand deliveryListCommandCopy = new DeliveryListCommand(DeliveryStatus.SHIPPED, null, null,
+            Sort.ASC);
+        assertEquals(deliveryListCommand, deliveryListCommandCopy);
+
+        // different types -> returns false
+        assertNotEquals(deliveryListCommand, new DeliveryViewCommand(1));
+
+        // null -> returns false
+        assertNotEquals(deliveryListCommand, null);
+
+        // different status -> returns false
+        DeliveryListCommand differentDeliveryListCommand = new DeliveryListCommand(DeliveryStatus.CANCELLED, null,
+            null, Sort.ASC);
+        assertNotEquals(deliveryListCommand, differentDeliveryListCommand);
+
+        // different sort -> returns false
+        differentDeliveryListCommand = new DeliveryListCommand(DeliveryStatus.SHIPPED, null, null, Sort.DESC);
+        assertNotEquals(deliveryListCommand, differentDeliveryListCommand);
+
+        // different customer id -> returns false
+        differentDeliveryListCommand = new DeliveryListCommand(DeliveryStatus.SHIPPED, 1, null, Sort.ASC);
+        assertNotEquals(deliveryListCommand, differentDeliveryListCommand);
```

Comment:
Would these be better if it was values against values rather than null against values

Reply:
Are you talking about `assertNotEquals(deliveryListCommand, null);`?

Reply:
Are you talking about assertNotEquals(deliveryListCommand, null);?